### PR TITLE
Add new edtion and product id for Open Liberty

### PR DESCRIPTION
**Pull request by idlewis**
To distinguish features from WebSphere Liberty features
#build

### CODE REVIEW

File: dev/com.ibm.ws.repository/src/com/ibm/ws/repository/resources/internal/AppliesToProcessor.java [hunk 1]
```diff
@@ -247,19 +245,34 @@ public static void validateEditions(AppliesToFilterInfo info, String appliesToHe
...
+     * Parse the given raw appliesTo header into a list
+     * of AppliesToEntry objects
      */
     public static List<AppliesToEntry> parseAppliesToEntries(String appliesTo) {
         List<AppliesToEntry> result = new ArrayList<AppliesToEntry>();
+        List<List<String>> parsedProducts = doParse(appliesTo);
+        for (List<String> parsedProduct : parsedProducts) {
+            AppliesToEntry entry = new AppliesToEntry();
+            for (String thing : parsedProduct) {
+                entry.add(thing);
```
Comment (Azquelt) [inline dev/com.ibm.ws.repository/src/com/ibm/ws/repository/resources/internal/AppliesToProcessor.java]: `thing` isn't a great name

`productChunk` as used above might be better.

Also, it's not obvious reading the code what the difference between an `AppliesToEntry` and an `AppliesToFilterInfo` is. I assume we need methods for both though?

File: dev/com.ibm.ws.repository/src/com/ibm/ws/repository/resources/internal/AppliesToProcessor.java [hunk 2]
```diff
@@ -247,19 +245,34 @@ public static void validateEditions(AppliesToFilterInfo info, String appliesToHe
...
+        return result;
+    }
 
+    /**
+     * Should be parsing a string that looks like
+     * "blah;blah;blah,foo;foo;foo,baz;baz;baz"
+     * into a list of lists, so something like
+     * <blah,blah,blah>,<foo,foo,foo>,<baz,baz,baz>
+     */
+    private static List<List<String>> doParse(String appliesTo) {
```
Comment (Azquelt) [inline dev/com.ibm.ws.repository/src/com/ibm/ws/repository/resources/internal/AppliesToProcessor.java]: Nice comment :+1:

File: dev/com.ibm.ws.repository/src/com/ibm/ws/repository/resources/internal/AppliesToProcessor.java [hunk 3]
```diff
@@ -33,10 +34,20 @@
  */
 public class AppliesToProcessor {
 
-    private final static List<String> allEditions = Arrays.asList("Liberty Core", "Base", "Express", "Developers", "ND", "z/OS");
+    /**
+     * Open liberty features with an edition of OPEN in the appliesTo can be
+     * installed on everything except CORE.
+     */
+    private final static List<String> nonCoreEditions = Arrays.asList("Base", "Express", "Developers", "ND", "z/OS");
```
[the remark this answers is not in the document]
Comment (idlewis) [reply dev/com.ibm.ws.repository/src/com/ibm/ws/repository/resources/internal/AppliesToProcessor.java]: that makes sense

**Comment by idlewis** [comment]
#build

**Comment by idlewis** [comment]
#build

**Comment by idlewis** [comment]
The most recent build is red due to one UI test failure, which is not related to this change

**Pull request merged**